apps/web/src/app/api/sync-logs/route.ts

```diff
@@ -0,0 +1,93 @@
+import { createClient } from '@tuturuuu/supabase/next/server';
+import { NextResponse } from 'next/server';
+
+export async function GET() {
+  const supabase = await createClient();
+
+  const {
+    data: { user },
+    error: authError,
+  } = await supabase.auth.getUser();
+
+  if (authError || !user) {
+    return NextResponse.json({ message: 'Unauthorized' }, { status: 401 });
+  }
+
+  const { data, error } = await supabase
+    .from('calendar_sync_dashboard')
+    .select(
+      'id, starttime, type, ws_id, triggered_by, status, endtime, events_inserted, events_updated, events_deleted'
+    )
+    .order('starttime', { ascending: true });
+
+  if (error)
+    return NextResponse.json(
+      { message: 'Error fetching sync logs' },
+      { status: 500 }
+    );
+
+  // Fetch all required data in batch
+  const workspaceIds = [...new Set(data.map((item) => item.ws_id))];
+  const userIds = [
+    ...new Set(data.map((item) => item.triggered_by).filter(Boolean)),
+  ];
```

Comment:
you should get workspace data and user data in the select query too, do not do what you're doing here as it's not scalable and will greatly affect the system stability.

e.g.
```tsx
.select('id, start_time, ..., workspaces!inner(id, name), users(id, display_name, avatar_url), ...'
```